### DISCUSSION TITLE
Only -7kb with RTDB

### DISCUSSION THREAD

**Discussion by ghost** [firebase/firebase-js-sdk#4377] [Q&A] [answered]
Tried the most basic demo for RTDB, bundled by https://github.com/vitejs/vite
The resulting js is only 7kb smaller. Is it normal?

from:
```js
import firebase from 'firebase/app'
import 'firebase/database'

const firebaseConfig = {
//
}

const db = firebase.initializeApp(firebaseConfig)
export default db
```

to:
```js
import { initializeApp } from 'firebase/app'
import { getDatabase } from 'firebase/database'

const firebaseConfig = {
//
}

const firebaseApp = initializeApp(firebaseConfig)
const db = getDatabase(firebaseApp)
export default db
```

**Comment by davideast** [comment, marked as answer]
This is currently the case with RTDB. We only provided the database instance as a tree-shakeable function. You'll notice the rest of the methods still chain off of the instance instead of requiring functions for operations. This isn't our long term plan but it's our current state of the alpha right now. I'll let @schmidt-sebastian detail more if he's free.

**Comment by ghost** [comment]
on the other hand, node_modules is ~35Mb smaller. That's a plus)

**Comment by schmidt-sebastian** [comment]
We are currently working on a tree-shakeable API of the RTDB SDK and hope to have some updates for you soon.
